dotCMS/src/main/java/com/dotcms/security/ContentSecurityPolicyUtil.java

```diff
@@ -0,0 +1,192 @@
+        if (isConfig()) {
+            String contentSecurityPolicyHeader = getContentSecurityPolicyHeader();
+
+            for (final Entry<String, ContentSecurityPolicyResolver> entry : contentSecurityPolicyResolvers.entrySet()) {
+                if (contentSecurityPolicyHeader.contains(entry.getKey())) {
+                    contentSecurityPolicyHeader = entry.getValue().headerResolver.resolve(contentSecurityPolicyHeader);
+                }
+            }
+
+            response.addHeader("Content-Security-Policy", contentSecurityPolicyHeader);
+        }
+    }
+
+    public static String apply(final String htmlCodeParam) {
+        String htmlCodeResult = htmlCodeParam;
+
+        if (isConfig()) {
+            String contentSecurityPolicyHeader = getContentSecurityPolicyHeader();
+
+            for (final Entry<String, ContentSecurityPolicyResolver> entry : contentSecurityPolicyResolvers.entrySet()) {
+                if (contentSecurityPolicyHeader.contains(entry.getKey())) {
+                    htmlCodeResult = entry.getValue().htmlCodeResolver
+                            .resolve(contentSecurityPolicyHeader, htmlCodeResult);
+                }
+            }
+        }
+
+        return htmlCodeResult;
+    }
+
+    private static String calculateNonceToScript(final String contentSecurityPolicyConfig,
+            final String htmlCode) {
+
+        final String nonce = getNonceFromCurrentRequest();
+        return htmlCode.replaceAll("<script", String.format("<script nonce='%s'",nonce));
+    }
+
+    private static String calculateNonceHeaderToScript(final String header) {
+        final String nonce = getNonceFromCurrentRequest();
+        return header.replace("{script-src nonce}", String.format("'nonce-%s'",nonce));
+    }
+
+    private static String calculateNonceToStyle(final String contentSecurityPolicyConfig,
+            final String htmlCode) {
+
+        final String nonce = getNonceFromCurrentRequest();
+        return htmlCode.replaceAll("<style", String.format("<style nonce='%s'",nonce));
+    }
+
+    private static String calculateNonceHeaderToStyle(final String header) {
+        final String nonce = getNonceFromCurrentRequest();
+        return header.replace("{style-src nonce}", String.format("'nonce-%s'",nonce));
+    }
+
+    private static String getNonceFromCurrentRequest() {
+        final HttpServletRequest request = HttpServletRequestThreadLocal.INSTANCE.getRequest();
+        final String nonce = (String) request.getAttribute(NONCE_REQUEST_ATTRIBUTE);
+        return nonce;
+    }
+
+    @FunctionalInterface
+    interface HeaderResolver {
+        String resolve(final String header);
+    }
+
+    @FunctionalInterface
+    interface HtmlCodeResolver {
+        String resolve(final String contentSecurityPolicyConfig, final String htmlCode);
+    }
+
+    private static class ContentSecurityPolicyResolver {
+        HeaderResolver headerResolver;
```

Comment:
seems such as this could be final

Reply:
done https://github.com/dotCMS/core/pull/21337/commits/bffbe06951439bc17ef47dea074b0d441d470cc1